certification/internal/engine/engine.go

```diff
@@ -279,3 +298,162 @@ func untar(dst string, r io.Reader) error {
+			DockerVersion: config.DockerVersion,
+			ImageID:       digest.String(),
+			Labels:        labels,
+			OS:            config.OS,
+			Size:          size,
+			//TODO: figure out how to get/convert this data from crane
+			UncompressedLayerSizes: layerSize,
+		},
+		RawConfig:         string(rawConfig),
+		SumLayerSizeBytes: sumLayerSizeBytes(layers),
+		//TODO: figure out how to get/convert this data from crane
+		UncompressedTopLayerId: "placeholder",
+	}
+
+	// calling MarshalIndent so the json file written to disk is human-readable when opened
+	certImageJson, err := json.MarshalIndent(certImage, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	fileName, err := artifacts.WriteFile(certification.DefaultCertImageFilename, string(certImageJson))
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrSaveFileFailed, err)
+	}
+
+	log.Tracef("image config written to disk: %s", fileName)
+
+	return nil
+}
+
+func writeRPMManifest(containerFSPath string) error {
+	db, err := rpmdb.Open(filepath.Join(containerFSPath, "var", "lib", "rpm", "Packages"))
+	if err != nil {
+		return err
+	}
+
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return err
+	}
+
+	// covert rpm struct to pxyis struct
+	rpms := make([]pyxis.RPM, 0, len(pkgList))
+	for _, packageInfo := range pkgList {
+
+		var bgName string
+		var endChop string
+		var srpmNevra string
+
+		// accounting for the fact that not all packages have a source rpm
+		if len(packageInfo.SourceRpm) > 0 {
+			bgName = packageInfo.SourceRpm[0:strings.LastIndex(strings.SplitAfter(packageInfo.SourceRpm, ".")[0], "-")]
+			endChop = strings.TrimPrefix(strings.TrimSuffix(regexp.MustCompile("(-[0-9].*)").FindString(packageInfo.SourceRpm), ".rpm"), "-")
+
+			srpmNevra = fmt.Sprintf("%s-%d:%s", bgName, packageInfo.Epoch, endChop)
+		}
+
+		rpm := pyxis.RPM{
+			Architecture: packageInfo.Arch,
+			Gpg:          "placeholder", //TODO: change this once GPG is implemented in go-rpm library
+			Name:         packageInfo.Name,
+			Nvra:         fmt.Sprintf("%s-%s-%s.%s", packageInfo.Name, packageInfo.Version, packageInfo.Release, packageInfo.Arch),
+			Release:      packageInfo.Release,
+			SrpmName:     bgName,
+			SrpmNevra:    srpmNevra,
+			Summary:      packageInfo.Summary,
+			Version:      packageInfo.Version,
+		}
+
+		rpms = append(rpms, rpm)
+	}
+
+	rpmManifest := pyxis.RPMManifest{
+		RPMS: rpms,
+	}
+
+	// calling MarshalIndent so the json file written to disk is human-readable when opened
+	rpmManifestJson, err := json.MarshalIndent(rpmManifest, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	fileName, err := artifacts.WriteFile(certification.DefaultRPMManifestFilename, string(rpmManifestJson))
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrSaveFileFailed, err)
+	}
+
+	log.Tracef("rpm manifest written to disk: %s", fileName)
+
+	return nil
+}
+
+func sumLayerSizeBytes(layers []v1.Layer) int64 {
+	var sum int64
+	for _, layer := range layers {
+		size, _ := layer.Size()
```

Comment:
seems like we are ignoring error here. Do we want to ignore the error? or fail the submit?

Reply:
Good catch, I forgot about this call. This has been fixed.